hazelcast/src/test/java/com/hazelcast/jet/JobStatusListenerTest.java

```diff
@@ -289,45 +291,39 @@ protected static void assertHasNoListenerEventually(String jobIdString, UUID reg
         });
     }
 
-    protected void testListener(JobConfig config, Object source, BiConsumer<Job, JobStatusLogger> test) {
-        testListener(source, (jet, p) -> jet.newJob(p, config), test);
+    protected void testListener(JobConfig config, DAG dag, BiConsumer<Job, JobStatusLogger> test) {
+        testListener(dag, (jet, p) -> jet.newJob(p, config), test);
     }
 
-    protected void testListener(Object source, BiConsumer<Job, JobStatusLogger> test) {
-        testListener(new JobConfig(), source, test);
+    protected void testListener(DAG dag, BiConsumer<Job, JobStatusLogger> test) {
```

Comment:
This chain of  `testListener` methods calling one another is not really readable.

If we had one method taking some parameters, even at the cost of _some_ repetition at the call site, it would be more readable.

Reply:
or use approach similar to `com.hazelcast.jet.core.test.TestSupport`